### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by wandergithub**
# OOP school library: preserve data

- On exit, app save all the data (books, people, and rentals) into JSON files.
- On startup of your app load all data from files.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -1,14 +1,62 @@
...
+      (0...rentals.length).each do |i|
+        rental = JSON.parse(rentals[i], create_additions: true)
+        @books.each { |book| book.rental << rental[0] if rental[0].book.title == book.title }
+        @people.each { |person| person.rental << rental[0] if rental[0].person.name == person.name }
+      end
+    end
+    # binding.pry
+    people_file.close
+    rental_file.close
+  end
```
Comment (DammyShittu) [inline app.rb]: - [x] Currently, your app saves the data to the files. However, you are expected to also handle scenarios when one or more storage JSON files are not present. When any file is not present, you are expected to create the file as well or handle it in some way but it should not throw errors.

  For example, if I remove the JSON files and run the app, it throws errors, but it is expected that such scenarios are handled in some way or those files are created if they are absent. Kindly implement that.

  rental-json

<hr />

-  [x] Additionally, when the files are present but empty, it throws errors as well. I think to resolve this scenario you can add an empty array if the file is empty, depending on your code.

  empty

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi team,

## Highlights

✔️ Data is saved on app exit

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (wandergithub) [reply app.rb]: Done 👍🏽

**Review by OLIPLICHE — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Linter is green (the last commit have a green check mark (✅)) 👏
README file is descriptive 👏
Commit messages are meaningful 👏 👏
Github flow is correctly used (Pull Request is made from the feature branch into the main branch) 👏
App working fine👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**